rest/admin_api_test.go

```diff
@@ -2270,45 +2270,40 @@ func TestHandlePutDbConfigWithBackticks(t *testing.T) {
 }
 
 func TestHandleDBConfig(t *testing.T) {
-	t.Skip("disabled pending CBG-1566")
 	if base.GTestBucketPool.NumUsableBuckets() < 2 {
 		t.Skipf("test requires at least 2 usable test buckets")
 	}
 
-	rt := NewRestTester(t, nil)
-
 	tb := base.GetTestBucket(t)
 
-	defer tb.Close()
+	rt := NewRestTester(t, &RestTesterConfig{TestBucket: tb})
 	defer rt.Close()
 
 	bucket := tb.GetName()
-	resource := fmt.Sprintf("/%s/", bucket)
-
-	// Create a database with no config
-	resp := rt.SendAdminRequest(http.MethodPut, resource, `{"num_index_replicas":0}`)
-	assertStatus(t, resp, http.StatusCreated)
-	assert.Empty(t, resp.Body.String())
+	dbname := "db"
+	resource := fmt.Sprintf("/%s/", dbname)
 
 	// Get database config before putting any config.
-	resp = rt.SendAdminRequest(http.MethodGet, resource, "")
+	resp := rt.SendAdminRequest(http.MethodGet, resource, "")
 	assertStatus(t, resp, http.StatusOK)
 	var respBody db.Body
-	assert.NoError(t, respBody.Unmarshal([]byte(resp.Body.String())))
+	assert.NoError(t, respBody.Unmarshal(resp.Body.Bytes()))
 	assert.Nil(t, respBody["bucket"])
-	assert.Equal(t, bucket, respBody["db_name"].(string))
+	assert.Equal(t, dbname, respBody["db_name"].(string))
 	assert.Equal(t, "Online", respBody["state"].(string))
 
 	// Put database config
-	resource = fmt.Sprintf("/%v/_config", bucket)
+	resource = resource + "_config"
 
 	// change cache size so we can see the update being reflected in the API response
 	dbConfig := &DbConfig{
+		BucketConfig: BucketConfig{Bucket: &bucket},
 		CacheConfig: &CacheConfig{
 			RevCacheConfig: &RevCacheConfig{
 				Size: base.Uint32Ptr(1337), ShardCount: base.Uint16Ptr(7),
 			},
 		},
+		NumIndexReplicas:   base.UintPtr(0),
```

Comment:
We can also set these, primarily to avoid the test creating indexes when we don't have them enabled.

```suggestion
		NumIndexReplicas:   base.UintPtr(0),
		EnableXattrs:       base.BoolPtr(base.TestUseXattrs()),
		UseViews:           base.BoolPtr(base.TestsDisableGSI()),
```